lib/connector.js

```diff
@@ -155,12 +155,12 @@ Connector.getNestedPropertyDefinition = function(definition, propPath) {
   const isArray = !isPropUndefined && Array.isArray(prop.type);
   const isFunction = !isPropUndefined && !isArray && typeof prop.type === 'function';
 
+  if (propPath.length === 1) return prop;
+
   if (isPropUndefined || (propPath.length > 1 && (isArray && prop.type.length === 0))) {
     throw new Error(g.f('Invalid property path'));
   }
```

Comment:
This method is to retrieve metadata. IMO, we should return `undefined` if no property definition exists for such path and have caller of this function to throw if necessary.